base/base/TL.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+#include <utility>
+#include "defines.h"
+#include "TypePair.h"
+
+/// General-purpose typelist. Easy on compilation times as it does not use recursion.
+template <class ...Args>
+struct TL { static constexpr size_t size = sizeof...(Args); };
```

Comment:
I'd prefer `TypeList` and it's considered to use full names in ClickHouse code base.

Reply:
Well, I'm fine with the full name, but, as for me, the only other meaning for TL is "time limit" which is no applicable here. I've searched ClickHouse code base, every usage of TL states clearly by context that a type is used.

If you want, I can change the class name (TypeList) but leave function names (e.g. TLConcat).

Reply:
> If you want, I can change the class name (TypeList) but leave function names (e.g. TLConcat).

Ok, let's do it.